posts/2021-01-22-jwtbuilder-redirects-21001.adoc

```diff
@@ -0,0 +1,162 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15337[Dynacache initialization issue when ID is missing]
++
+An external user discovered that the initialization of a `distributedMap` fails with a NullPointerException if the `id` element is not present in the `distributedMap` configuration.
+A clearer message should have been displayed to indicate the required `id` element is missing.  To address this the `distributedMap` definition has been modified to mark the `id` element as required.
++
+The configuration runtime will now issue an error message: 
++
+`CWWKG0058E: The element distributedMap with the unique identifier default-0 is missing the required attribute id.`
++
+and the `distributedMap` will not be put into service. This new behaviour is introduced in Open Liberty 21.0.0.1. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14925[OAuth user registry lookups may use incorrect custom cache key]
++
+In previous releases, a flaw existed where a previously authenticated user’s `Subject` might not be found in the authentication cache during an OAuth authentication flow, when using a custom user registry. The cache key used to retrieve user claims from the authentication cache was based on the realm and username, but the correct cache key might be a combination of the OAuth provider name and the OAuth token object itself. The behavior has been corrected and the appropriate cache key should now be used.
++
+For more information visit the link:https://openliberty.io/docs/21.0.0.1/reference/feature/oauth-2.0.html[Open Liberty Documentation].
+
+* Add HTTP/2 IOException for misbehaving client error case
++
+It was previously possible for the HTTP/2 channel to throw a `NullPointerException` when it attempted to write out HTTP headers on a connection that had been terminated due to a connection error. Beginning in Open Liberty 21.0.0.1, the HTTP/2 channel will now throw a more informative `IOException` for this scenario. Read more about link:https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/cwlp_servlet40_http2.html[Open Liberty's support for HTTP/2].
+
+* CONTAINER_NAME env variable is not reflected in logstashCollector-1.0
++
+Starting from Open Liberty 20.0.0.9, JSON logs created by the `logstashCollector-1.0` feature do not properly reflect the value set for the environment variable `CONTAINER_NAME`. The value set for `CONTAINER_NAME` should be reflected in the `serverName` field of the JSON logs, but the default server name from `wlp.server.name` is shown instead. This behaviour has been corrected for Open Liberty 21.0.0.1, visit the link:https://openliberty.io/docs/21.0.0.1/forwarding-logs-logstash.html[Logstash Collector Documentation] for more information.
```

Comment:
Change https://openliberty.io/docs/21.0.0.1/forwarding-logs-logstash.html to a non-versioned url:
https://openliberty.io/docs/latest/forwarding-logs-logstash.html

Reply:
Updated.